parser/src/lib.rs

```diff
@@ -256,6 +256,7 @@ mod test {
                     };
                     eprint!("\t{sign}{change}");
                 }
+                eprintln!("The following string was re-parsed:\n{orig_asm_to_string}");
```

Comment:
Somehow, the newly added test fails the re-parse test! It seems to forget the `Main` min and max degree:

```
The following string was re-parsed:
use std::machines::binary::ByteBinary as ByteBinary;
use std::machines::large_field::binary::Binary as Binary;
machine Main {
    reg pc[@pc];
    reg X0[<=];
    reg X1[<=];
    reg X2[<=];
    reg A;
    ByteBinary byte_binary;
    Binary binary(byte_binary, 8, 8);
    instr and X0, X1 -> X2 link ~> X2 = binary.and(X0, X1){  }
    function main {
    A <== and(2863311530, 2863311530);
    A <== and(4294967295, 4294967295);
    return;
    }
}

thread 'test::parse_write_reparse_asm' panicked at parser/src/lib.rs:260:17:
parsed and re-parsed ASTs differ for file: /Users/georg/coding/powdr/test_data/asm/block_machine_exact_number_of_rows.asm
```

Reply:
Ohhh

Reply:
Fixed thanks to #2079 😍